### DISCUSSION TITLE
Qt5's "Qt Script" module discarded in Qt6; impact on Stellarium scripts ?

### DISCUSSION THREAD

**Discussion by josevarelab** [Stellarium/stellarium#1620] [General]
Dear Stellarium folks,

You mention (I think it's Georg) in your article dated 2021-03-24, published in the "Journal of Skyscape Archaeology", that you are planning to upgrade to Qt6.

As Qt6 is not supporting/taking-over the old Qt5's "Qt Script" module hosting the JavaScriptCore engine (which is the current ECMAScript provider), would this have an impact on Stellarium scripting capabilities ?

Thanks in advance for any info on the subject. Best regards.

**Comment by gzotti** [comment]
Sure. This is one of the two big changes we face with Qt6, the other being changes in the graphics subsystems where OpenGL is no longer the common unifying platform. 

Fabien had tried Qt's replacement solution for JS, but found it unsuitable. Qt5.14 introduced some flow control functionalities which may solve the problem and may allow us to keep ECMAscripting. But so far nobody has started migrating this part to Qt6.

**Comment by josevarelab** [comment]
Hi Georg,

Many thanks for the prompt reply.

Sorry to re-insist but let me rephrase the question (which may be a shared concern by other users as well) :

By upgrading to Qt6, is there a possibility that current user-written Stellarium scripts won't work anymore and/or will need to be rewritten ?

I know that you will take into consideration other pros and cons like graphic-related aspects, but my question is specific to Stellarium scripts : will they continue to work as before after a (potential) migration to Qt6 ?

Txs in advance for the clarification. Best regards.

**Reply by gzotti** [reply]
It's Difficult to Make Predictions, Especially About the Future ...

Yes, it is possible that scripting will change. The recent finds appear positive enough that there is a chance to keep ECMAscript and not convert e.g. to some entirely different scripting language.  However, I think nobody on this planet can give a clear yes or no answer before somebody actually starts migrating to the replacement infrastructure. I hope only little will change. That is hope as in "hope", not knowing anything for certain or even being able to tell the future. I only know for sure lots of work is still lying ahead.

**Comment by josevarelab** [comment]
Hi Georg,

Txs again for your prompt reply.

However I am a bit disappointed by your reply. My question is simple, there is no trap behind, nothing to do with predictions.

I know there is always risk when migrating Qt*.dll components from one minor version to the other (e.g. 5.9.x.x to the current 5.12.6.0).

However, another situation is a total break like the one regarding the old StratoScript engine mentioned in your webpage "http://www.stellarium.org/scripts.html" :
As of version 0.10.1, Stellarium uses the QtScript engine. The previously used StratoScript is no longer available. ...

My understanding, at this stage (I may be wrong, that's why I'm asking you), is that a migration/upgrade to Qt6 is more likely to be a StratoScript-like scenario as Qt6 is dropping Qt5's "Qt Script" module which is the component servicing Stellarium scripts.

I understand that Qt6 may be a better solution than Qt5-unsupported features, but if this migration/upgrade will make current scripts obsolete it would be great to let us know for not putting too much effort in writing scripts that will need to be re-written after the upgrade.

Once again thanks in advance for any info you may provide me with and best regards.

**Reply by gzotti** [reply]
Sorry, but which part of 

HOW SHOULD I KNOW THIS?

don't you understand?

If you know how to do the change, implement the change in Qt5.15 and are able to avoid a change for users' scripts, then there will be no change.

**Comment by josevarelab** [comment]
My apologies Georg, I didn't want to be unpleasant.

Sincerely many thanks for the great work you and the reduced team have been doing in developing this great tool, good continuation.

Best regards.
